packages/backend/src/outgoing_payment/service.test.ts

```diff
@@ -0,0 +1,771 @@
+          account: 'http://wallet.example/bob',
+          amount: 56,
+          assetCode: 'XRP',
+          assetScale: 9,
+          description: 'description!',
+          received: 0,
+          expiresAt: new Date(Date.now() + 60 * 1000).toISOString(),
+          ilpAddress: credentials.ilpAddress,
+          sharedSecret: credentials.sharedSecret.toString('base64')
+        })
+        .persist()
+      await knex.raw('TRUNCATE TABLE "outgoingPayments" RESTART IDENTITY')
+    }
+  )
+
+  afterEach((): void => {
+    nock.cleanAll()
+    jest.useRealTimers()
+    jest.restoreAllMocks()
+
+    for (const plugin of Object.values(plugins)) {
+      // Plugins must be cleaned up, otherwise ilp-plugin-http can leak http2 connections.
+      expect(plugin.isConnected()).toBe(false)
+    }
+    plugins = {}
+  })
+
+  afterAll(
+    async (): Promise<void> => {
+      await appContainer.shutdown()
+      await truncateTables(knex)
+    }
+  )
+
+  describe('get', (): void => {
+    it('returns undefined when no payment exists', async () => {
+      await expect(outgoingPaymentService.get(uuid())).resolves.toBeUndefined()
+    })
+  })
+
+  describe('create', (): void => {
+    it('creates an OutgoingPayment', async () => {
+      const payment = await outgoingPaymentService.create({
+        superAccountId,
+        paymentPointer: 'http://wallet.example/paymentpointer/bob',
+        amountToSend: BigInt(123),
+        autoApprove: false
+      })
+      expect(payment.state).toEqual(PaymentState.Inactive)
+      expect(payment.intent).toEqual({
+        paymentPointer: 'http://wallet.example/paymentpointer/bob',
+        amountToSend: BigInt(123),
+        autoApprove: false
+      })
+      expect(payment.superAccountId).toBe(superAccountId)
+      await expect(
+        connectorService.getAccountBalance(payment.sourceAccount.id)
+      ).resolves.toEqual({ balance: BigInt(0) })
+      expect(payment.sourceAccount.code).toBe('USD')
+      expect(payment.sourceAccount.scale).toBe(9)
+      expect(payment.destinationAccount).toEqual({
+        scale: 9,
+        code: 'XRP',
+        url: 'http://wallet.example/paymentpointer/bob'
+      })
+
+      const payment2 = await outgoingPaymentService.get(payment.id)
+      if (!payment2) throw 'no payment'
+      expect(payment2.id).toEqual(payment.id)
+    })
+  })
+
+  describe('processNext', (): void => {
+    describe('Inactive→', (): void => {
+      it('Ready (FixedSend)', async (): Promise<void> => {
+        const paymentId = (
+          await outgoingPaymentService.create({
+            superAccountId,
+            paymentPointer: 'http://wallet.example/paymentpointer/bob',
+            amountToSend: BigInt(123),
+            autoApprove: false
+          })
+        ).id
+        const payment = await processNext(paymentId, PaymentState.Ready)
+        if (!payment.quote) throw 'no quote'
+        expect(payment.quote.timestamp).toBeInstanceOf(Date)
+        expect(
```

Comment:
nit: space out the portions of a test into the three phases

setup

act

assert

It just makes it easier to read